miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -239,91 +234,106 @@ public BoxUse getUse(long id) throws IOException {
 
   @Override
   public long save(Box box) throws IOException {
-    setChangeDetails(box);
+    box.setChangeDetails(authorizationManager.getCurrentUser());
     if (box.getStorageLocation() != null) {
       box.setStorageLocation(storageLocationService.get(box.getStorageLocation().getId()));
     } else {
       box.setStorageLocation(null);
     }
     loadChildEntities(box);
-    if (box.getId() == AbstractBox.UNSAVED_ID) {
+    if (!box.isSaved()) {
       return saveNewBox(box);
     } else {
-      Box original = boxStore.get(box.getId());
-      authorizationManager.throwIfNotWritable(original);
-      logStorageChange(box, original);
-      validateChange(box, original);
-      applyChanges(box, original);
+      Box managed = boxStore.get(box.getId());
+      authorizationManager.throwIfNotWritable(managed);
+      logStorageChange(box, managed);
+      validateChange(box, managed);
+      applyChanges(box, managed);
       StringBuilder message = new StringBuilder();
 
       // get persisted version of new box contents before change
-      List<BoxableId> ids = box.getBoxables().values()
+      List<BoxableId> ids = box.getBoxPositions().values()
           .stream()
-          .map(b -> b.getId())
+          .map(b -> b.getBoxableId())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)